src/queries/metadata.ts

```diff
@@ -0,0 +1,52 @@
+import { gql } from '@apollo/client';
+
+export const METADATA = gql`
+    query {
+        allMetadata {
+            id
+            name
+            description
+            published
+            source
+            updated
+            tags
+            visualisations {
+                axes {
+                    x {
+                        name
+                    }
+                    y {
+                        name
+                    }
+                }
+            }
+        }
+    }
+`;
+
+export type MetadataEntry = {
+    id: string;
+    name: string;
+    description: string;
+    published: string;
+    source: string;
+    updated: string;
+    tags: string[];
+    visualisations: [
+        {
+            type: string;
+            axes: {
+                x: {
+                    name: string;
+                };
+                y: {
+                    name: string;
+                };
+            };
+        },
+    ];
```

Comment:
This is a common error which can give you some very funky behaviour later on.

What you're saying here is that `visusalisations` is an array of one (not zero, not more) objects of the type you nested in the array.

What you're looking for is either `Array<{ ... }>` or `{ ... }[]`. This is one reason for why some opt for the former, as it looks less like the mishap here.

Reply:
Good catch 🤽🏽‍♂️